### PR TITLE
Improve testing against distro and podman-next

### PR DESCRIPTION
**Pull request by inknos**
1. Split upstream into distro/pnext test runs

A new CI run targets tests that are exclusive for podman-next. /upstream plans are now split into /distro and /pnext to be explicit against which podman version is used in the tests. Also, python version used are now called base_python and all_python, to be more self explainatory.

2. Add pytest marker `pnext` to filter tests

Tests that should run against podman-next are now skipped by default in tox runs. They can be enabled with --pnext and filtered in using `-m pnext`. These two options should be used together since `pnext` and `non-pnext` tests are thought to be incompatible.

3. Split `test_container_mounts` to test a breaking change and update docs

The scenario `test_container_mounts` was split to demonstrates the usage of the marker against an upstream breaking change. CONTRIBUTING.md now reflects the new feature and explains how to use tox locally to leverage on this feature.

4. Add manual trigger on packit

Since the tests cover a corner case and serve the purpose of testing
unreleased features of libpod (which is slightly out of the scope of
podman-py) the testing jobs are optional and should run for certain PRs
only. The command to run the pnext tests is the following.

/packit test --labels {pnext,podman_next}

Fixes: https://github.com/containers/podman-py/issues/547

cc: @lsm5

### CODE REVIEW

File: .packit.yaml [hunk 1]
```diff
@@ -130,30 +131,38 @@ jobs:
...
-    tmt_plan: /upstream/all
+    identifier: distro-fedora-all
+    tmt_plan: /distro/all_python
     packages: [python-podman-fedora]
     targets:
       - fedora-all
+
+  # run tests for the rawhide python version using podman-next packages
+  - job: tests
+    trigger: pull_request
```
Comment (lsm5) [inline .packit.yaml:142]: I guess this should be changed to `ignore` unless we're ok with them failing often.

**Review by lsm5 — APPROVED** [review]
LGTM apart from the `podman-next-fedora-base` failing often. Left a comment about that.

File: .packit.yaml (hunk 1 above)
Comment (lsm5) [reply .packit.yaml:142]: Alternatively, if you use this: https://packit.dev/posts/manual-triggering#triggering-test-jobs-based-on-labels-and-identifiers , you could run the default set of tests in all cases and run the podman-next test only when the pomdman-next / pnext label is added to the PR. That can be a manual action. When the label is not present, that test will simply not run and will show a rather misleading `waiting for RPM build to succeed` status and just not run. Maybe better than a failed test though.

Comment (inknos) [reply .packit.yaml:142]: that is perfect!

Comment (inknos) [reply .packit.yaml:142]: they should be passing now, though, so I'll fix them first

**Comment by inknos** [comment]
@lsm5 all is good now (except for cirrus which is red because it uses `podman-next`)

as said privately, this is a problem that we cannot get around that easily for cirrus. disabling the copr, is surely a quick fix, but it would prevent us from testing upstream changes

File: tests/main.fmf [hunk 2]
```diff
@@ -24,6 +24,12 @@ require:
...
 
+    /base_python_pnext:
+        tag: [ pnext ]
+        summary: Run all tests on the base python version and podman-next
+        test: cd .. && make tests-ci-base-python-podman-next
+        duration: 5m
+
     /all_python:
         tag: [ matrix ]
         summary: Run all tests for all ptyhon versions available
```
Comment (lsm5) [inline tests/main.fmf:35]:
```suggestion
        summary: Run all tests for all python versions available
```

Nit

**Review by lsm5 — APPROVED** [review]
/lgtm

**Review by lsm5 — APPROVED** [review]
/lgtm
/hold

**Review by lsm5 — APPROVED** [review]
/lgtm
